Feature 199 add home lambdas spec in home lambdas

jessesillman opened the pull request:
## What has been done

### 1. Added the Home Lambdas API Spec

- The updated `home-lambdas-api-spec` has been added as a submodule to `home-lambdas`.
- The spec is used to generate TS models via OpenAPI generator,  resulting in the `src/generated/homeLambdasModels` directory.

### 2. Automated Model Generation

- Updated `package.json`scripts, as following:
  - `postinstall` now runs `build-home-lambdas` models to automatically generate models after installing dependencies.
  - The `build-home-lambdas-models` script uses OpenAPI Generator to create models from the spec.
- After running `npm run build-home-lambdas-models`, all models should be available in `src/generated/homeLambdasModels`.

### 3. Applied the Generated Models to Handlers

- Updated the handler files to import and use the generated models for return types.
- Replaced `any` types from `/types`folder with spec-defined models (e.g. `YearlyVacationDays`).
- Comments have been added in places where there are typing conflicts or mismatches with the new spec.

### 4. Disabled Building of Deprecated/Inactive Lambdas

- Edited the function files to prevent building (trello, memo, slack, pipedrive)
- **Note**: This is controlled by the environment variable `SKIP_INACTIVE_LAMBDAS=true`, which must be set in the `.env` file.

### Notes for Review

- The build process should now generate only the models, similar how it is done for Severa. This ensures that all lambda can use spec-defined return types without unnecessary API code.
- Due to some conflicts and type mismatches between the already existing model and the new generated models, some old models (e.g. `ArticelModel` in `wiki-documentation` folder) are used for now.
- These mismatches should now be documented in the corresponding handler files, and the response is cast to the OpenAPI type for compatibility.
- These areas might need further refactoring to fully align with the new spec.

jessesillman committed:
chore(spec): add home-lambdas-API-spec submodule

jessesillman committed:
First commit

jessesillman committed:
Adding home lambda specs in handlers

jessesillman committed:
Updates

jessesillman committed:
Minor update

jessesillman committed:
Added the new models in on-call

jessesillman committed:
Create questionnaire update, cleaning

JuusoHeikkinen02 commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -56,12 +56,12 @@ export const createSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareM
       image: data.image,
       description: data.description,
-      review: data.review,
+      review: data.review !== undefined ? data.review : "",
```
```suggestion
      review: data.review ?? "",
```

JuusoHeikkinen02 reviewed (commented):
approved, left one comment.

StudVladimir commented on src/functions/keycloak/update-user-vacation/handler.ts:
```diff
@@ -1,8 +1,11 @@
 import { CreateKeycloakApiService } from "src/database/services/keycloak-api-service";
 import type { APIGatewayProxyHandlerV2 } from "aws-lambda";
-import type { VacationDays,VacationDayEntry } from "src/types/keycloak/vacations";
+// types folder will no longer needed
```
if it's no longer in use, don't leave comment

StudVladimir commented on src/functions/keycloak/update-user-vacation/handler.ts:
```diff
@@ -1,8 +1,11 @@
+//import type { VacationDays,VacationDayEntry } from "src/types/keycloak/vacations";
+import { YearlyVacationDays } from "src/generated/homeLambdasModels/model/yearlyVacationDays";
 
+// vacationDays matches the OpenAPI spec
```
unnecessary comment

jessesillman committed:
Small upadtes

StudVladimir commented on src/functions/keycloak/update-user-vacation/handler.ts:
```diff
@@ -46,12 +48,11 @@ const updateVacationHandler: APIGatewayProxyHandlerV2 = async (event) => {
+     * This format helps with easy lookup and updating in Keycloak user attributes.
  *
  * Example:
  *   formatValue("2024", 5) => "2024:005"
```
Fix tabulation for TSDoc

StudVladimir replied:
For TSDoc below also

jessesillman committed:
Tabulation fix in hanler

jessesillman committed:
Conflicts resolved

Jdallos commented on package.json:
```diff
@@ -8,8 +8,9 @@
-    "build-severa-client": "openapi-generator-cli generate -i severa-API-spec/swagger.yaml -o ./src/generated/severaClient -c generator-config.json -g typescript-node"
+    "postinstall": "npm run build-severa-client && npm run build-home-lambdas-models",
+    "build-severa-client": "openapi-generator-cli generate -i severa-API-spec/swagger.yaml -o ./src/generated/severaClient -c generator-config.json -g typescript-node",
+    "build-home-lambdas-models": "openapi-generator-cli generate -i home-lambdas-API-spec/swagger.yaml -o ./src/generated/homeLambdasModels -g typescript-node --global-property models,modelDocs=false,apis=,apiDocs=false,supportingFiles="
```
Some of the syntax here, e.g. "apis="
Looks a bit weird, please check

Alexander-K-byte replied:
```suggestion
    "build-home-lambdas-models": "openapi-generator-cli generate -i home-lambdas-API-spec/swagger.yaml -o ./src/generated/homeLambdasModels -g typescript-node --global-property models,modelDocs=false,modelTests=false"
```

Alexander-K-byte replied:
Once you initialize global-property flag, it only includes what you specified, however docs and tests default true for models and apis.

Jdallos commented on .gitmodules:
```diff
@@ -1,3 +1,7 @@
+[submodule "home-lambdas-API-spec"]
+	path = home-lambdas-API-spec
+	url = https://github.com/Metatavu/home-lambdas-API-spec
+	branch = develop
```
```suggestion
```

Looks like in most projects we don't specify the branch here, I suspect this is pipline related so we can have multiple environments. As such lets not do it here

Jdallos commented on src/functions/keycloak/find-user/handler.ts:
```diff
@@ -16,13 +18,22 @@ const findUserHandler: APIGatewayProxyHandler = async (
-    const userById = await api.findUser(id);
-    if (!userById) {
+    const userByIdRaw = await api.findUser(id);
+
```
```suggestion
```

Jdallos commented on src/functions/keycloak/find-user/handler.ts:
```diff
@@ -16,13 +18,22 @@ const findUserHandler: APIGatewayProxyHandler = async (
         body: JSON.stringify({ error: "User not found" }),
       };
     }
+    // Map the CustomKeycloakProfile to User model
```
```suggestion
```

Jdallos commented on src/functions/keycloak/update-user-vacation/handler.ts:
```diff
@@ -1,8 +1,9 @@
-import type { VacationDays,VacationDayEntry } from "src/types/keycloak/vacations";
+import { YearlyVacationDays } from "src/generated/homeLambdasModels/model/yearlyVacationDays";
 
+// vacationDays matches the OpenAPI spec
```
```suggestion
```

Jdallos commented on src/functions/keycloak/update-user-vacation/handler.ts:
```diff
@@ -11,6 +12,7 @@ const keycloakApiService = CreateKeycloakApiService();
  * @param event - API Gateway event containing userId and vacationDays in body
  * @returns Response message as JSON string
  */
+// NOTE: Returned vacationDays matches the OpenAPI spec: object with years as keys and YearlyVacationDays as values.
```
```suggestion
```

Jdallos commented on src/functions/memo-management/drive-memos/get-content-pdf/index.ts:
```diff
@@ -1,16 +1,21 @@
-        authorizer: {
-          name: "homeKeycloakAuthorizer"
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
No sure we should use an env for this, rather can't we just set isInactive to true directly here?

Jdallos commented on src/functions/memo-management/drive-memos/get-memos-pdf/index.ts:
```diff
@@ -1,10 +1,15 @@
-  events: [
-    {
-      httpApi: {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
```
Same as previously and for all other inacvitve endpoints, please update. The comment can remain however

Jdallos commented on src/functions/memo-management/drive-memos/get-summary-memo-pdf/index.ts:
```diff
@@ -1,9 +1,14 @@
-  events: [
-    {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/drive-memos/get-translated-memo-pdf/index.ts:
```diff
@@ -1,9 +1,14 @@
-  events: [
-    {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/trello-cards/comment-trello-card/index.ts:
```diff
@@ -1,16 +1,21 @@
-        authorizer: {
-          name: "homeKeycloakAuthorizer"
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/trello-cards/create-trello-card/index.ts:
```diff
@@ -1,10 +1,14 @@
-    {
-      httpApi: {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/trello-cards/delete-trello-card/index.ts:
```diff
@@ -1,10 +1,14 @@
-    {
-      httpApi: {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/trello-cards/get-board-members/index.ts:
```diff
@@ -1,10 +1,14 @@
-    {
-      httpApi: {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: inactive).
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/memo-management/trello-cards/get-trello-cards/index.ts:
```diff
@@ -1,10 +1,15 @@
-    {
-      httpApi: {
+// NOTE: This lambda is marked as active in the OpenAPI spec (x-status: active)
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/add-interest-to-deal/index.ts:
```diff
@@ -1,16 +1,22 @@
-  ],
-}
\ No newline at end of file
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same as before

Jdallos commented on src/functions/pipedrive/add-interest-to-lead/index.ts:
```diff
@@ -1,11 +1,17 @@
-    {
-      httpApi: {
-        method: 'PATCH',
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/find-deal-by-id/index.ts:
```diff
@@ -1,10 +1,16 @@
-  events: [
-    {
-      httpApi: {
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/find-lead-by-id/index.ts:
```diff
@@ -1,9 +1,15 @@
-  handler: `${handlerPath(__dirname)}/handler.main`,
-  events: [
-    {
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/list-deals/index.ts:
```diff
@@ -1,9 +1,15 @@
-  handler: `${handlerPath(__dirname)}/handler.main`,
-  events: [
-    {
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/list-leads/index.ts:
```diff
@@ -1,9 +1,15 @@
-  handler: `${handlerPath(__dirname)}/handler.main`,
-  events: [
-    {
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/remove-interest-from-deal/index.ts:
```diff
@@ -1,7 +1,13 @@
 
-export default {
-  handler: `${handlerPath(__dirname)}/handler.main`,
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/pipedrive/remove-interest-from-lead/index.ts:
```diff
@@ -1,9 +1,15 @@
-  handler: `${handlerPath(__dirname)}/handler.main`,
-  events: [
-    {
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/questionnaire/create-questionnaire/handler.ts:
```diff
@@ -29,20 +38,26 @@ export const createQuestionnaireHandler: ValidatedEventAPIGatewayProxyEvent<type
+    /**
+     * NOTE:
+     * There may be a type mismatch: OpenAPI model defines passedUsers as string[],
+     * but the service expects number[]. This may cause issues if user IDs are not numbers.
```
Interesting and I'm a little concerned this might cause issues. Make sure to test this once this is merged it might be that it breaks something but we can investigate it further then to figure out what this should be.

Jdallos commented on src/functions/questionnaire/update-questionnaire/handler.ts:
```diff
@@ -36,18 +37,22 @@ const updateQuestionnaireHandler: ValidatedEventAPIGatewayProxyEvent<typeof ques
 
   try {
-    const updatedQuestionnaire = await questionnaireService.updateQuestionnaire(questionnaireUpdates);
+    // NOTE: Type mismatch in passedUsers field (number[] vs string[]). For now, just cast via unknown as per spec.
```
Again lets leave this as is but check it out after merger, might cause issues

Jdallos commented on src/functions/slack-user-avatar/index.ts:
```diff
@@ -1,10 +1,14 @@
-    {
-      httpApi: {
+// NOTE: This lambda is marked as inactive in the OpenAPI spec (x-status: active)
+const isInactive = process.env.SKIP_INACTIVE_LAMBDAS === "true";
```
same again

Jdallos commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -16,7 +16,7 @@ const softwareService = new SoftwareService(docClient);
  */
-export const createSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareModel> = async (event) => {
+export const createSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareRegistry> = async (event) => {
   console.log('Received event:', JSON.stringify(event));
```
remove console.logs

Jdallos commented on src/functions/software-registry/find-software/handler.ts:
```diff
@@ -33,10 +34,19 @@ export const findSoftwareHandler: APIGatewayProxyHandler = async (event: APIGate
+        ...software,
+        // NOTE: TS types for status for status differ between SoftwareModel and SoftwareRegistry.
+        // This cast ensures compatibility with the OpenAPI spec model without modifying generated types.
+        status: software.status as unknown as SoftwareRegistry["status"],
```
What is the difference? Casting does not actually fix any type mismatches and so this might cause some issues, if unsure lets leave it as it was originally.

Jdallos commented on src/functions/software-registry/find-software/handler.ts:
```diff
@@ -33,10 +34,19 @@ export const findSoftwareHandler: APIGatewayProxyHandler = async (event: APIGate
+        createdAt: software.createdAt ? new Date(software.createdAt) : undefined,
+        lastUpdatedAt: software.lastUpdatedAt ? new Date(software.lastUpdatedAt) : undefined
+      };
+      console.log('Software found:', fixedSoftware);
```
R3emove console.logs

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
+ * 
  * @returns Response object with status code and body.
  */
+
```
```suggestion
```

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
  */
+
 export const listSoftwareHandler: APIGatewayProxyHandler = async () => {
   console.log('Received request to list software entries');
```
Remove console.log

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
   console.log('Received request to list software entries');
 
   try {
     console.log('Fetching software list from DynamoDB');
```
remove

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
     const softwareList = await softwareService.listSoftware();
 
-    console.log('Software list retrieved successfully:', softwareList);
+    // Map SoftwareModel[] to SoftwareRegistry[]
```
```suggestion
```

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
+    const softwareRegistryList: SoftwareRegistry[] = softwareList.map((software) => ({
+      ...software,
+      // Cast status to match SoftwareRegistry type
+      status: software.status as unknown as SoftwareRegistry["status"],
```
What is this type difference, the types need to match we can't just cast to a type that is not true

Jdallos commented on src/functions/software-registry/list-software/handler.ts:
```diff
@@ -11,19 +12,34 @@ const softwareService = new SoftwareService(docClient);
+      lastUpdatedAt: software.lastUpdatedAt ? new Date(software.lastUpdatedAt) : undefined,
+    }));
+
+    console.log('Software list retrieved successfully:', softwareRegistryList);
```
remove

Jdallos commented on src/functions/software-registry/update-software/handler.ts:
```diff
@@ -63,53 +54,51 @@ export const updateSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareM
 
-    const updatedSoftware = await softwareService.updateSoftware(id, updatedSoftwareData);
+    // NOTE: Type assertion used to bypass type mismatch between SoftwareRegistry and SoftwareModel.
+    // review is optional in SoftwareRegistry but required in SoftwareModel.
```
We can always update the models (used for the DB) to match the spec. It might be that is the appropriate solution here .

Jdallos commented on src/functions/software-registry/update-software/handler.ts:
```diff
@@ -63,53 +54,51 @@ export const updateSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareM
-    const updatedSoftware = await softwareService.updateSoftware(id, updatedSoftwareData);
+    // NOTE: Type assertion used to bypass type mismatch between SoftwareRegistry and SoftwareModel.
+    // review is optional in SoftwareRegistry but required in SoftwareModel.
+    // status types also differ
```
As mentioned already lets clarify the difference and figure out what to do. Would wasnt to avoid doing updatedSoftwareData as any

Jdallos commented on src/functions/vacation-request/create-vacation-request/handler.ts:
```diff
@@ -4,12 +4,17 @@ import { v4 as uuidv4 } from "uuid";
  * @returns Response object with status code
+ * 
+ * Type mismatches between VacationRequestModel and VacationRequest:
+ * - VacationRequestModel may be missing 'message' property required by VacationRequest.
```
add message to the model?

Jdallos commented on src/functions/vacation-request/create-vacation-request/handler.ts:
```diff
@@ -4,12 +4,17 @@ import { v4 as uuidv4 } from "uuid";
+ * 
+ * Type mismatches between VacationRequestModel and VacationRequest:
+ * - VacationRequestModel may be missing 'message' property required by VacationRequest.
+ * - 'createdAt', 'updatedAt', 'startDate', 'endDate' are 'string' here, but VacationRequest expects 'Date'.
```
set as date then?

Jdallos commented on src/functions/vacation-request/find-vacation-request/handler.ts:
```diff
@@ -32,9 +38,13 @@ const findVacationRequestHandler: APIGatewayProxyHandler = async (event: APIGate
       }
     }
 
+    // NOTE: Type mismatches in fields like 'message', 'type', 'createdAt', 'updatedAt' (string vs Date).
```
If spec says its a Date it should probably be a date, how do we handle DB dates for the other models in use e.g. software registry or Questionnaires?

Jdallos commented on src/functions/vacation-request/list-vacation-request/handler.ts:
```diff
@@ -2,14 +2,19 @@ import type { APIGatewayProxyHandler } from "aws-lambda";
 /**
  * Labmda for listing all vacation requests from DynamoDB.
+ * Type mismatches between VacationRequestModel and VacationRequest:
+ * - VacationRequestModel may be missing 'message' property required by VacationRequest.
```
As mentinoed in previous comments

Jdallos commented on src/functions/wiki-documentation/create-article/handler.ts:
```diff
@@ -15,6 +16,14 @@ const articleService = new ArticlesApiService(docClient);
+ * 
+ * Type mismatches between ArticleModel and Article:
+  * @remarks
+ * - ArticleModel uses 'string' for date fields ('createdAt', 'lastUpdatedAt', 'lastReadAt'), but Article expects 'Date'.
```
Clarify but probably we should store dates in a consistant way, we probably want to avoid changing things unnecessaryil y and I think it's fine to have the DB store date as string but the backend to expect and parse it to Date.

Jdallos commented on src/functions/wiki-documentation/create-article/handler.ts:
```diff
@@ -15,6 +16,14 @@ const articleService = new ArticlesApiService(docClient);
+ * Type mismatches between ArticleModel and Article:
+  * @remarks
+ * - ArticleModel uses 'string' for date fields ('createdAt', 'lastUpdatedAt', 'lastReadAt'), but Article expects 'Date'.
+ * - ArticleModel requires 'id', but Article has 'id' as optional.
```
This is fine comment can be removed. We always sotre with an id but sometimes e.g. on creation an article might not have an id until it is created/ stored

Jdallos commented on src/functions/wiki-documentation/create-article/handler.ts:
```diff
@@ -15,6 +16,14 @@ const articleService = new ArticlesApiService(docClient);
+  * @remarks
+ * - ArticleModel uses 'string' for date fields ('createdAt', 'lastUpdatedAt', 'lastReadAt'), but Article expects 'Date'.
+ * - ArticleModel requires 'id', but Article has 'id' as optional.
+ * - ArticleModel does not have a 'content' field, but Article requires 'content'.
```
Investigate, we probably need to add. Are we not storing content already? Through the UI it looks liek we are

Jdallos commented on src/functions/wiki-documentation/update-article/handler.ts:
```diff
@@ -15,6 +16,11 @@ const articleService = new ArticlesApiService(docClient);
+ * - ArticleModel requires 'id', but Article has 'id' as optional.
+ * - ArticleModel does not have a 'content' field, but Article requires 'content'.
+ * - ArticleModel may have required fields that are optional in Article (e.g. 'description', 'coverImage', 'tags', 'readBy').
+ * - Property names and structures may differ for some fields.
```
See previous comment about articles and these types

jessesillman committed:
Update src/functions/keycloak/find-user/handler.ts

Co-authored-by: Josh Dallos <78350882+Jdallos@users.noreply.github.com>

JuusoHeikkinen02 committed:
Merge remote-tracking branch 'origin/develop' into feature-199-add-home-lambdas-spec-in-home-lambdas

JuusoHeikkinen02 committed:
Made most of the suggested changes that were mostly code cleanup, further suggested changes will be made with Jesse

JuusoHeikkinen02 committed:
made suggested changes

JuusoHeikkinen02 committed:
Removed comments about the issues that have been fixed, Changed handlers to use generated models

Alexander-K-byte committed:
updated package.json with raw url for swagger file to genererate  homeLambdasModel.

Alexander-K-byte committed:
updated submodule, replaced raw URL in package,json with proper path to swagger file.

Alexander-K-byte committed:
updated software.ts to remove enum type; updated handler so that  object is built on SoftwareModel

Alexander-K-byte committed:
id no longer optional

Alexander-K-byte committed:
api-gateway, organizded imports:
handler.ts, used questionaireSchema for validation of properties: resolved possibly infinite recursion of body by using schema alias: Organized imports.

Alexander-K-byte committed:
Double quotes used instead of backticks for template literal

Alexander-K-byte committed:
updated db model for questionnaire to  accept string array as this is what is being used instead of number array

Alexander-K-byte committed:
Checked all handler files for questionnaire functions. organized imports and corrected errors

Alexander-K-byte committed:
removed unused import;

Alexander-K-byte committed:
Finishing changes to organize imports and minor issues.

Alexander-K-byte committed:
conversion done from string to date to avoid issues. (quick fix)

Alexander-K-byte committed:
organized imports

Alexander-K-byte committed:
minor code change and organized imports

jessesillman committed:
Added type imports, small fixes

Alexander-K-byte committed:
organize imports split  imports into type / value and imported twice. Placed in single import statement and organized to resolve SonarQube issue.

Alexander-K-byte committed:
Static Alias implemented; commented out currently unused imports and  variable.

Alexander-K-byte committed:
organized imports; commented out currently unused imports;

Jdallos commented on src/functions/questionnaire/create-questionnaire/handler.ts:
```diff
@@ -1,26 +1,37 @@
+import questionnaireSchema from "src/schema/questionnaire";
+import { v4 as uuidv4 } from "uuid";
+
+export const createQuestionnaireHandler: ValidatedEventAPIGatewayProxyEvent<Questionnaire> = async (
```
I don't think the change of typeof questionnaireSchema to Questionnaire will work here due to how it's used in the ValidatedEventAPIGatewayProxyEvent.

I would suggest setting it back to typeof questionnaireSchema

Jdallos replied:
MIssing TSDOcs

Jdallos commented on src/functions/questionnaire/find-questionnaire/handler.ts:
```diff
@@ -17,33 +18,33 @@ const findQuestionnaireHandler: APIGatewayProxyHandler = async (event: APIGatewa
-
-    const quizById = await questionnaireService.findQuestionnaire(id);
-
+    // NOTE: Type mismatch in passedUsers field (number[] vs string[]).
```
What about updating the DB model to be string?

Jdallos replied:
Remove comment

Jdallos commented on src/functions/questionnaire/list-questionnaire/handler.ts:
```diff
@@ -1,27 +1,31 @@
-      await questionnaireService.listQuestionnaires();
+    const allQuestionnaires = await questionnaireService.listQuestionnaires();
+
+    // Cast each item to Questionnaire to match the OpenAPI spec
```
Again could we change the DB type to be string for passedUsers?

Jdallos replied:
Remove comment

Jdallos commented on src/functions/questionnaire/update-questionnaire/handler.ts:
```diff
@@ -1,25 +1,26 @@
 /**
  * Lambda function to update a questionnaire
- *
+ * QuestionnaireBody computed once, prevents possible infinite recurse
```
Is this comment necessrary?

Jdallos commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -1,88 +1,88 @@
     const authData = getAuthDataFromToken(event);
     if (!authData || !authData.sub) {
-      console.log('User is not authenticated');
+      console.log("User is not authenticated");
```
This should be console.error, although as we have the return message perhaps this can be removed

Jdallos commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -1,88 +1,88 @@
 
     const loggedUserId = authData.sub;
-    console.log('Logged User ID (sub claim):', loggedUserId);
+    console.log("Logged User ID (sub claim):", loggedUserId);
```
Remove console.log

Jdallos commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -1,88 +1,88 @@
+      lastUpdatedBy: loggedUserId
     };
-    console.log('Creating new software with data:', newSoftware);
+    console.log("Creating new software with data:", newSoftware);
```
Remove console.log

Jdallos commented on src/functions/software-registry/create-software/handler.ts:
```diff
@@ -1,88 +1,88 @@
     // Create new software entry in the database
     const createdSoftware = await softwareService.createSoftware(newSoftware);
-    console.log('Software created successfully:', createdSoftware);
+    console.log("Software created successfully:", createdSoftware);
```
remove console.log

Jdallos commented on src/functions/software-registry/delete-software/handler.ts:
```diff
@@ -1,50 +1,52 @@
+export const deleteSoftwareHandler: APIGatewayProxyHandler = async (
+  event: APIGatewayProxyEvent
+) => {
+  console.log("Received event:", JSON.stringify(event));
```
Remove consol.log

Jdallos commented on src/functions/software-registry/delete-software/handler.ts:
```diff
@@ -1,50 +1,52 @@
   }
   const { id } = event.pathParameters || {};
-  console.log('Path parameter (id):', id);
+  console.log("Path parameter (id):", id);
```
Remove

Jdallos commented on src/functions/software-registry/delete-software/handler.ts:
```diff
@@ -1,50 +1,52 @@
 
   if (!id) {
-    console.log('Missing ID in path parameters.');
+    console.log("Missing ID in path parameters.");
```
remove

Jdallos commented on src/functions/software-registry/delete-software/handler.ts:
```diff
@@ -1,50 +1,52 @@
 
   try {
-    console.log('Looking up existing software with id:', id);
+    console.log("Looking up existing software with id:", id);
```
remove

Jdallos commented on src/functions/software-registry/delete-software/handler.ts:
```diff
@@ -1,50 +1,52 @@
 
     if (!existingSoftware) {
-      console.log('Software not found.');
+      console.log("Software not found.");
```
remove

Jdallos commented on src/functions/software-registry/find-software/handler.ts:
```diff
@@ -10,46 +11,51 @@ const softwareService = new SoftwareService(docClient);
  */
 export const findSoftwareHandler: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {
-  console.log('Received event:', JSON.stringify(event));
+  console.log("Received event:", JSON.stringify(event));
```
remove

Jdallos commented on src/functions/software-registry/find-software/handler.ts:
```diff
@@ -10,46 +11,51 @@ const softwareService = new SoftwareService(docClient);
 
   const { id } = event.pathParameters || {};
-  console.log('Path parameter (id):', id);
+  console.log("Path parameter (id):", id);
```
remove

Jdallos commented on src/functions/software-registry/find-software/handler.ts:
```diff
@@ -10,46 +11,51 @@ const softwareService = new SoftwareService(docClient);
 
   try {
-    console.log('Finding software with id:', id);
+    console.log("Finding software with id:", id);
```
remove

Jdallos commented on src/functions/software-registry/update-software/handler.ts:
```diff
@@ -16,98 +17,80 @@ const softwareService = new SoftwareService(docClient);
+export const updateSoftwareHandler: ValidatedEventAPIGatewayProxyEvent<SoftwareRegistry> = async (
+  event
+) => {
+  console.log("Received event:", JSON.stringify(event));
```
remove

JuusoHeikkinen02 committed:
Changed database models to match the spec and deleted console logs

Jdallos commented on src/functions/questionnaire/create-questionnaire/handler.ts:
```diff
@@ -1,26 +1,37 @@
-  const { title, description, questions, tags, passedUsers, passScore } = event.body;
-  
+
+  // Parse body if needed
```
remove

Jdallos commented on src/functions/questionnaire/create-questionnaire/handler.ts:
```diff
@@ -1,26 +1,37 @@
+    body = JSON.parse(event.body);
+  }
+
+  // Validate with questionnaireSchema
```
remove

Alexander-K-byte committed:
TSDoc comment re-added
